src/Tizen.Peripheral/Tizen.Peripheral/I2c.cs

```diff
@@ -0,0 +1,151 @@
+﻿/*
+* Copyright (c) 2020 Samsung Electronics Co., Ltd All Rights Reserved
+*
+* Licensed under the Apache License, Version 2.0 (the License);
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an AS IS BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using NativeI2c = Interop.Peripheral.I2c;
+
+namespace Tizen.Peripheral
+{
+    /// <summary>
+    /// The class allows applications to communicate via i2c platform's bus.
+    /// </summary>
+    /// <privilege>http://tizen.org/privilege/peripheralio</privilege>
+    public class I2c : IDisposable
+    {
+        /// <summary>
+        /// Native handle to I2c.
+        /// </summary>
+        protected IntPtr _handle;
+
+        private I2c(IntPtr handle)
+        {
+            _handle = handle;
+        }
+
+        /// <summary>
+        /// Opens an I2C slave device.
+        /// </summary>
+        /// <param name="bus">The I2C bus number that the slave device is connected.</param>
+        /// <param name="address">The address of the slave device.</param>
+        /// <returns>The I2c object.</returns>
+        public static I2c Open(int bus, int address)
+        {
+            var ret = NativeI2c.Open(bus, address, out IntPtr handle);
+            if (ret != Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            return new I2c(handle);
+        }
+
+        /// <summary>
+        /// Closes an I2C slave device.
+        /// </summary>
+        public void Close()
+        {
+            var ret = NativeI2c.Close(_handle);
+            if (ret != Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+        }
+
+        /// <summary>
+        /// Reads the bytes data from the I2C slave device.
+        /// </summary>
+        /// <param name="size">The size of array to read.</param>
+        /// <returns>The array of bytes.</returns>
+        public byte[] Read(int size)
```

Comment:
The form of using the buffer inside the read method prevents the reuse of the buffer.
Also, when an Exception occurs, the created Buffer is just discarded.
You also need a Read method that takes the Buffer itself as a method argument.